jul-ecs-formatter/src/main/java/co/elastic/logging/jul/MdcSupplier.java

```diff
@@ -0,0 +1,79 @@
+/*-
+ * #%L
+ * Java ECS logging
+ * %%
+ * Copyright (C) 2019 - 2020 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.logging.jul;
+
+import org.slf4j.MDC;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public interface MdcSupplier {
+    Map<String, String> getMDC();
+
+    enum Resolver {
+        INSTANCE;
+
+        MdcSupplier resolve() {
+            try {
+                Class.forName("org.slf4j.MDC");
+
+                // The SLF4J API dependency does not contain an MDC binder by itself, the MDC binders come from an SLF4j
+                // implementation. When no MDC bindings are available calls to MDC.put will be ignored by slf4j.
+                // That is why we want to ensure that the StaticMDCBinder exists
+                Class.forName("org.slf4j.impl.StaticMDCBinder");
+                return Available.INSTANCE;
+            } catch (Exception e) {
+                return Unavailable.INSTANCE;
+            } catch (LinkageError e ) {
+                return Unavailable.INSTANCE;
+            }
+        }
+    }
+
+    enum Available implements MdcSupplier {
+        INSTANCE;
+
+        @Override
+        public Map<String, String> getMDC() {
+            Map<String, String> copyOfContextMap = MDC.getCopyOfContextMap();
+            if (copyOfContextMap != null ) {
+                return copyOfContextMap;
+            }
+            return new HashMap<String, String>();
+        }
+    }
+
+    enum Unavailable implements MdcSupplier {
+        INSTANCE;
+
+        Unavailable() {
+        }
+
+        @Override
+        public Map<String, String> getMDC() {
+            return new HashMap<String, String>();
```

Comment:
return `Collections.emptyMap()`

Reply:
Done. I somehow assumed that this was not possible as I first had `return Objects.requireNonNullElse(copyOfContextMap, Collections.emptyMap());` but maven wouldn't compile that. 

On a side note, is it intended that maven compiles to java 6 (legacy support or something else)? My idea hints me towards code style improvements that don't compile. e.g. multi catch blocks like this: `catch (Exception | LinkageError e) {
                return Unavailable.INSTANCE;
            }`